fix: Throw exception on invalid base64

Oinkling opened the pull request:
Fixed JWT::urlsafeB64Decode not actually throwing an InvalidArgumentException on invalid base64 characters, despite doc comment saying that it does.

bshaffer commented:
It looks like it currently throws an `InvalidArgumentException` but you want it to throw a `UnexpectedValueException`?

As this would be a breaking change, wouldn't it be better to just reflect the doc comment to match what's happening, rather than alter the existing behavior?

Oinkling commented:
The current behavior of `urlsafeB64Decode` is to call `base64_decode` using the default value (`false`) for the `$strict` parameter.  
This will result in PHP just ignoring any characters that are not a part of base64 and as such never actually fail, despite the documentation saying that it will.  
Since a valid JWT will never contain any characters that are outside of base64url I don't think this should be considered a breaking change as those tokens would most likely have triggered one of the other exceptions that validates the token anyway.

In regards to the changing I do in `decode` with `InvalidArgumentException` to `UnexpectedValueException`  
As explained above the current behavior is to ignore invalid characters, the fact that `urlsafeB64Decode` even throws an `InvalidArgumentException` now is because of my change, so that change in itself doesn't break much.  
The reason I change them to `UnexpectedValueException` is because every other validation error in `decode` uses that exception, so any try/catch around it will still work as normal, introducing no breaking changes, only adding more explanations as to what went wrong.

I would had had `urlsafeB64Decode` simply throw an `UnexpectedValueException` and then not having to do all the converting up in `decode` if it weren't for the documentation stating that it should throw an `InvalidArgumentException`

Oinkling commented:
I have added tests to demonstrate the error I described

However, while working on that I realized that my solution would still allow invalid base64URL through, as long as it is valid base64. I have also fixed that now.

bshaffer commented on src/JWT.php:
```diff
@@ -111,14 +111,22 @@ public static function decode(
+        try {
+            $headerRaw = static::urlsafeB64Decode($headb64);
+        } catch (InvalidArgumentException $e) {
+            throw new UnexpectedValueException('Unable to decode header');
```
This method also can throw `InvalidArgumentException`, so I am not sure why we need to change the exception between methods
